app/services/case_extend_sar_deadline_service.rb

```diff
@@ -0,0 +1,89 @@
+class CaseExtendSARDeadlineService
+  attr_reader :result, :error
+
+  def initialize(user, kase, extension_days, reason)
+    @user = user
+    @case = kase
+    @extension_days = extension_days
+    @extension_deadline = new_extension_deadline(@extension_days.to_i)
+    @reason = reason
+    @result = :incomplete
+  end
+
+  def call
+    ActiveRecord::Base.transaction do
+      if validate_params
+        @case.state_machine.extend_sar_deadline!(
+          acting_user: @user,
+          acting_team: @user.team_for_case(@case),
+          final_deadline: @extension_deadline,
+          original_final_deadline: @case.external_deadline,
+          message: message
+        )
+
+        @case.extend_deadline!(@extension_deadline)
+        @result = :ok
+      end
+    end
+    @result
+  rescue => err
+    Rails.logger.error err.to_s
+    Rails.logger.error err.backtrace.join("\n\t")
+    @error = err
+    @result = :error
+  end
+
+  private
+
+  def new_extension_deadline(extend_by)
+    if @case.deadline_extended?
+      @case.external_deadline + extend_by.days
+    else
+      @case.initial_deadline + extend_by.days
+    end
+  end
+
+  def message
+    "#{@reason}\nDeadline extended by #{@extension_days} days"
+  end
+
+  def validate_params
```

Comment:
It's not completely obvious to me that this method returns a falsey value when there is an error, which is required by its caller.

Reply:
Very good point. Have re-written this code however have not back ported the same to extend_for_pit_service as that is well beyond the scope of this ticket.